pandas/core/arrays/integer.py

```diff
@@ -371,14 +376,28 @@ def __getitem__(self, item):
             return self._data[item]
         return type(self)(self._data[item], self._mask[item])
 
-    def _coerce_to_ndarray(self):
+    def _coerce_to_ndarray(self, dtype=None, na_value=libmissing.NA):
         """
         coerce to an ndarary of object dtype
         """
-
         # TODO(jreback) make this better
-        data = self._data.astype(object)
-        data[self._mask] = self._na_value
+        if dtype is None:
+            dtype = object
+        elif is_float_dtype(dtype) and na_value is libmissing.NA:
+            # XXX: Do we want to implicitly treat NA as NaN here?
+            # We should be deliberate in this decision.
+            na_value = np.nan
+
+        if is_integer_dtype(dtype):
+            if not self.isna().any():
+                return self._data.astype(dtype)
+            else:
+                raise ValueError(
```

Comment:
this doesn't make any sense, by definition we have a nullable type. why does it matter if it has NAs which we are not caring about

Reply:
This is for IntegerArray.astype("int"), so converting to a NumPy array.

Apparently, the expected behavior is for that to succeed if there are no missing values, and raise if there are any (that's the behavior on master).

Reply:
> this doesn't make any sense, by definition we have a nullable type. why does it matter if it has NAs which we are not caring about

Sorry, I don't understand your comment. Do you mean we should always raise? Why don't we care about NAs?

(given your comments, the confusion is maybe this: this function is about conversion of IntegerArray -> numpy array (for astype, `__array__`, ..), not about coercing an array-like -> IntegerArray)

Reply:
ahh ok. yeah maybe this could be more clear.